pkg/services/ngalert/provisioning/alert_rules.go

```diff
@@ -87,6 +86,45 @@ func (service *AlertRuleService) CreateAlertRule(ctx context.Context, rule model
 	return rule, nil
 }
 
+// UpdateRuleGroup will update the interval for all rules in the group.
+func (service *AlertRuleService) UpdateRuleGroup(ctx context.Context, orgID int64, namespaceUID string, ruleGroup string, interval int64) error {
+	if err := service.validateRuleGroupInterval(interval); err != nil {
+		return err
+	}
+	return service.xact.InTransaction(ctx, func(ctx context.Context) error {
```

Comment:
How does this work because looking at the code for `ListAlertRules` and `UpdateAlertRules`, the former isn't queried in a transaction and the latter has its own transaction which is committed independent of this?

Reply:
I agree about `ForUpdate`, but what I don't understand is what `service.xact.InTransaction` does when a second transaction is created in `UpdateAlertRules`? The `service.xact.InTransaction` seems like it can be removed?

Reply:
ah, I get it now. the transaction is not created it is re-used because the context that is provided to the db store command contains the reference to the opened transaction

Reply:
Oh wow, I didn't expect that!